parsedate/parsedate.go

```diff
@@ -0,0 +1,167 @@
+package parsedate
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func NextDate(now time.Time, date string, repeat string) (string, error) {
+	parseDate, err := time.Parse("20060102", date)
+	if err != nil {
+		return "", fmt.Errorf("Bad date format %s", date)
+	}
+
+	switch {
+	case strings.HasPrefix(repeat, "d "):
+		part := strings.Split(repeat, " ")
+		if len(part) != 2 {
+			return "", errors.New("Bad format")
```

Comment:
![image](https://github.com/user-attachments/assets/bcdc52bc-9bf2-48cc-873d-f32b55900c07)
```suggestion
			return "", errors.New("bad format")
```